bundles/org.eclipse.ui.ide/src/org/eclipse/ui/ide/undo/ResourceDescription.java

```diff
@@ -31,69 +27,100 @@
  * This class is not intended to be extended by clients.
  *
  * @since 3.3
- *
+ * @deprecated
```

Comment:
Please add a note to this annotation telling which class should be used instead.